pdu/ShortMessage.go

```diff
@@ -134,28 +119,18 @@ func (c *ShortMessage) GetMessage() (st string, err error) {
 
 // GetMessageWithEncoding returns (decoded) underlying message.
 func (c *ShortMessage) GetMessageWithEncoding(enc data.Encoding) (st string, err error) {
-	if len(c.messageData) == 0 {
-		return
+	if len(c.messageData) > 0 {
+		st, err = enc.Decode(c.messageData)
 	}
-
-	f, t := 0, len(c.messageData)
-
-	// skip if UDL is present
-	f = c.udHeader.UDHL()
-	if f >= t {
-		err = errors.ErrUDHTooLong
-		return
-	}
-
-	st, err = enc.Decode(c.messageData[f:t])
 	return
 }
 
-// Split one short message and split into multiple short message, with UDH
+// split one short message and split into multiple short message, with UDH
 // according to 33GP TS 23.040 section 9.2.3.24.1
-// NOTE: Split() will return array of length 1 if data length is still within the limit
+//
+// NOTE: split() will return array of length 1 if data length is still within the limit
 // The encoding interface can implement the data.Splitter interface for ad-hoc splitting rule
-func (c *ShortMessage) Split() (multiSM []*ShortMessage, err error) {
+func (c *ShortMessage) split() (multiSM []*ShortMessage, err error) {
```

Comment:
Unexport to prevent wrong usage. In user's example, he invokes: `submitSM.Message.Split`

While the correct one is: `submitSM.Split()`